microsoft/testsuites/kselftest/kselftest.py

```diff
@@ -123,15 +123,17 @@ def _install(self) -> bool:
                 self.node.os, "kselftests in LISA does not support this os"
             )
 
+        # Create the kselftest installation directory
+        # for both tar file source and distro build.
+        mkdir = self.node.tools[Mkdir]
+        mkdir.create_directory(self._kself_installed_dir.as_posix())
```

Comment:
It needs to create directory in Tar tool. So, when Tar is used in other places, the non-exist dirs can be created too.

Reply:
From my debug, Tar tool directory is the same as the parent of _kself_installed_dir. As I tried to re-use the current code, I just took advantage of the implementation of non-tar tool code and shared it among these two routes. I can change that if it causes issues to other test cases.

Reply:
From our design principles, tools should complete their goal with robust behavior. If the dest dir doesn't exist, it needs to create the dir.

Reply:
sure, I will make the change